### DISCUSSION TITLE
Consider rethinking NX and related packages bumping

### DISCUSSION THREAD

**Discussion by algirdaspaskevicius** [cortip/isomera#130] [Ideas]
Simply bumping NX version and related dependencies will sooner or later reach a stage where the NX CLI and the monorepo itself will no longer function as expected.

Any NX and its dependencies bumping should be done via NX CLI. A command `nx migrate latest` (or specific NX version) should be ran firstly and then `nx migrate --run-migrations` after that, which will bump NX and related package versions.

Also, sometimes, NX version bumping may require manual actions, such as deleting `workspaces.json` after a specific NX version update or changing executors from `@nrwl` to `@nx`...

After running the `nx migrate --run-migrations` command, the generated `migrations.json` file can be deleted.

You can read more at https://nx.dev/core-features/automate-updating-dependencies

**Comment by vygandas** [comment]
@algirdaspaskevicius , good point. Thank you for bringing this up! I will adjust dependabot and look into versions and if there was and what damage was done.

**Comment by vygandas** [comment]
@algirdaspaskevicius do you think nx itself is counted in? :D https://github.com/cortip/isomera/pull/95

**Comment by algirdaspaskevicius** [comment]
@nx @nrwl and all sub packages @nx/*  @nrwl/* should be excluded from automated bumping. The `nx migrate latest` will bump the versions.

**Comment by vygandas** [comment]
So "nx" is ok to auto-bump? I mean there's package named differently and looks like it's core package for everything else.

**Comment by algirdaspaskevicius** [comment]
I see. Package `nx` should also be excluded as the migration command will update the nx version.
